File: src/main/java/com/kamikazejam/datastore/base/exception/CacheException.java
package com.kamikazejam.datastore.base.exception;

import com.kamikazejam.datastore.base.Cache;

@SuppressWarnings("unused")
public class CacheException extends Exception {

    public CacheException(Cache<?, ?> cache) {
        super("C: [" + cache.getName() + "] exception");
    }

    public CacheException(String message, Cache<?, ?> cache) {
        super("C: [" + cache.getName() + "] exception: " + message);
    }

    public CacheException(String message, Throwable cause, Cache<?, ?> cache) {
        super("C: [" + cache.getName() + "] exception: " + message, cause);
    }

    public CacheException(Throwable cause, Cache<?, ?> cache) {
        super("C: [" + cache.getName() + "] exception: ", cause);
    }

    public CacheException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace, Cache<?, ?> cache) {
        super("C: [" + cache.getName() + "] exception: " + message, cause, enableSuppression, writableStackTrace);
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/base/exception/CacheException.kt
package com.kamikazejam.datastore.base.exception

import com.kamikazejam.datastore.base.Cache

@Suppress("unused")
open class CacheException : Exception {
    constructor(cache: Cache<*, *>) : super("C: [" + cache.name + "] exception")

    constructor(message: String, cache: Cache<*, *>) : super("C: [" + cache.name + "] exception: " + message)

    constructor(
        message: String,
        cause: Throwable?,
        cache: Cache<*, *>
    ) : super("C: [" + cache.name + "] exception: " + message, cause)

    constructor(cause: Throwable?, cache: Cache<*, *>) : super("C: [" + cache.name + "] exception: ", cause)

    constructor(
        message: String,
        cause: Throwable?,
        enableSuppression: Boolean,
        writableStackTrace: Boolean,
        cache: Cache<*, *>
    ) : super("C: [" + cache.name + "] exception: " + message, cause, enableSuppression, writableStackTrace)
}
